packages/data-provider/src/services/submittedDataService.ts

```diff
@@ -73,24 +136,82 @@ const service = (dependencies: BaseDependencies) => {
 				paginationOptions,
 				filterSql,
 			);
-			if (!recordsPaginated) {
+
+			if (recordsPaginated.length === 0) {
 				return fetchDataErrorResponse(PAGINATION_ERROR_MESSAGES.NO_DATA_FOUND);
 			}
 
 			const totalRecords = await getTotalRecordsByCategoryIdAndOrganization(categoryId, organization, filterSql);
 
 			logger.info(
 				LOG_MODULE,
-				`Retrieved '${recordsPaginated?.length}' Submitted data on categoryId '${categoryId}' organization '${organization}'`,
+				`Retrieved '${recordsPaginated.length}' Submitted data on categoryId '${categoryId}' organization '${organization}'`,
 			);
 
 			return {
-				data: parseSubmittedData(recordsPaginated),
+				data: recordsPaginated,
 				metadata: {
 					totalRecords,
 				},
 			};
 		},
+
+		deleteSubmittedDataBySystemId: async (
+			systemId: string,
+			dryRun: boolean,
+			reason: string,
+			userName: string,
+		): Promise<SubmittedDataResponse[]> => {
+			const { getSubmittedDataBySystemId } = submittedDataRepo;
+			const { getDictionaryById } = dictionaryRepo;
+			const { mapRecordsSubmittedDataResponse } = submittedUtils(dependencies);
+
+			// get SubmittedData by SystemId
+			const submittedData = await getSubmittedDataBySystemId(systemId);
+
+			if (!submittedData) {
+				throw new BadRequest(`No Submitted data found with systemId '${systemId}'`);
+			}
+
+			logger.info(LOG_MODULE, `Found Submitted Data with system ID '${systemId}'`);
+
+			// create array with records to be updated
+			const recordsToUpdate: SubmittedData[] = [submittedData];
+
+			// get dictionary
+			const dictionary = await getDictionaryById(submittedData.lastValidSchemaId);
+
+			if (!dictionary) {
+				throw new BadRequest(`Dictionary not found`);
+			}
+
+			// get dictionary relations
+			const dictionaryRelations = getDictionarySchemaRelations(dictionary);
+
+			const recordDependency = await searchDirectDependents(dictionaryRelations, submittedData);
+
+			if (recordDependency && recordDependency.length > 0) {
+				recordsToUpdate.push(...recordDependency);
+			}
+
+			if (dryRun === false) {
+				// Execute soft deletion on Submitted Data
+				const submittedDataIds = recordsToUpdate.map((records) => records.id);
+				const updatedRecords = await submittedDataRepo.updateMany(submittedDataIds, {
+					comment: `Soft-Delete reason: ${reason}`,
```

Comment:
Will this property ever contain information other than this message? Cause if not we should just call it `deleteReason`.

I'm wondering if we should revisit adding a "RecordChangeHistory" table where we can record "update" and "delete" actions.

Reply:
renaming this column from `comment` to `deleteReason`

Reply:
An approach could be to have a table like "RecordChangeHistory" along with a trigger so AFTER an update INSERT into  RecordChangeHistory table. All this should be implemented on a separate PR for separation of concerns